### PR TITLE
Assignment8

### PR DESCRIPTION
**Pull request by KennethWhite**
Collaboration with @Cwhite534

Had significant issues with our code analysis using Directory.Build.props, so that is a work in progress. Also we switched from using dependency injection for the logging to using the Serilog logger which may need to be changed back.

### CODE REVIEW

**Comment by grantdwoods** [comment]
> Collaboration with @Cwhite534
> 
> Had significant issues with our code analysis using Directory.Build.props, so that is a work in progress. Also we switched from using dependency injection for the logging to using the Serilog logger which may need to be changed back.

Can I ask why you would decide to switch back?

File: SecretSanta/src/SecretSanta.Api/Program.cs [hunk 1]
```diff
@@ -32,7 +32,9 @@ public static void Main(string[] args)
             Serilog.Debugging.SelfLog.Enable(msg => Debug.WriteLine(msg));
 
             Log.Logger = new LoggerConfiguration()
+                .MinimumLevel.Debug()
```
Comment (grantdwoods) [inline SecretSanta/src/SecretSanta.Api/Program.cs]: Nice! Have you guys discussed or considered a scenario where Verbose would be necessary? 
I would have to see this in action before having an intelligent conversation about it...

File: SecretSanta/src/SecretSanta.Api/Program.cs [hunk 2]
```diff
@@ -32,7 +32,9 @@ public static void Main(string[] args)
...
                 .ReadFrom.Configuration(Configuration)
+                .WriteTo.SQLite(Configuration.GetConnectionString("LogConnection"))
                 .Enrich.WithProperty("App Name", "SecretSanta.Api")
```
Comment (grantdwoods) [inline SecretSanta/src/SecretSanta.Api/Program.cs]: 👍

**Comment by KennethWhite** [comment]
Collaboration with @Cwhite534.

@MarkMichaelis you should have received invites or permission to view our Azure pages.

Kenny's Application Insights: [Link](https://portal.azure.com/#@eagles.ewu.edu/resource/subscriptions/1c0f852d-9636-4391-bbe9-aa80b76c78e2/resourceGroups/CSCD496Winter-rg/providers/microsoft.insights/components/CSCD496Winter/overview)
Casey's Application Insights: [Link](https://portal.azure.com/#@eagles.ewu.edu/resource/subscriptions/5994359a-38ce-459b-8f48-95a1712a8958/resourceGroups/CW-SecretSanta-rg/providers/microsoft.insights/components/CW-SecretSanta-Insights/overview)

We are still working on setting up the configuration as you had specified, our logging does not go to a database in production.

File: SecretSanta/Directory.Build.props [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project>
+  <PropertyGroup>
+    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)SecretSanta.ruleset</CodeAnalysisRuleSet>
```
Comment (MarkMichaelis) [inline SecretSanta/Directory.Build.props]: Great use of a build property!!!

**Pull request closed without merging**